### DISCUSSION TITLE
2.17.1-2

### DISCUSSION THREAD

**Discussion by bb-qq** [bb-qq/r8152#378] [Announcements]
⚠️If you want to report a problem or get support, **please fill out the template and create a ticket in [Issues](https://github.com/bb-qq/r8152/issues)** instead of posting in the forum.

* Improved performance especially on older platforms. (beta)
  * In an earlier version, the size of the RX buffer was reduced to improve stability on older platforms.
  * Since that time, the Realtek base driver and firmware have been updated, so I reverted this change at this time.
  * Please **let me know if stability has decreased significantly** since this release. I will change to reduce the RX buffer only for that platform again.
  
Note:

* The installation will fail the first time. After that, run the following command from the SSH terminal:
   `sudo install -m 4755 -o root -D /var/packages/r8152/target/r8152/spk_su /opt/sbin/spk_su`
* Selecting a proper version is needed depending on your DSM version.
  * DSM 7.2 and above
    * use packages with the suffix `_7.2`.
  * DSM 7.0 and DSM 7.1
    * use packages with the suffix `_7.1`.
  * DSM 6.2.x
    * use packages with the suffix `_6.2`.

<hr /><em>This discussion was created from the release <a href='https://github.com/bb-qq/r8152/releases/tag/2.17.1-2'>2.17.1-2</a>.</em>

**Comment by hhessel** [comment]
There aren't any _7.2 packages in this release or is this intentional due to the fact it's for "older platforms"?

**Reply by Sjekke** [reply]
I would say 7.2 are the new platforms 😅, but I agree where are they?

**Reply by bb-qq** [reply]
Oops, I added packages for DSM 7.2.
(I forgot to keep the release as a draft until the build is complete..)

**Reply by Sjekke** [reply]
👍

**Reply by DarkBlade123** [reply]
Nice work, fast and stable on DS224+

**Reply by ektorasdj** [reply]
Hello,
I cant begin to thank you for developing these drivers.
DS720+
Club 3D CAC-142
DSM 7.2.4-69057 Update 4
Installed on top of 2.17.1-1 without any errors.
Speeds are full 290mb/s (stable 290mb/s on transfers to NVMe storage volume)
MTU 9000
I have them installed almost 3 days and no issue yet.

**Comment by popcornmaster69** [comment]
I updated to this version 3 days ago on my DS718+. It's been running for 72hrs non-stop. The old driver would have been self disconnected by now and require a full reboot of the system in order to reconnect. I'm not sure it's the usb nic(asustor 2.5G) I use or its the old driver. So far so good. I'll let you know if I encounter same problem in the future. Great thank for your contribution.

**Comment by ChaG06** [comment]
Hello,
This is the first time that MTU 9000 is stable. With all previous versions, I had to keep it at 1500.
Thanks for this update. 😀

DS1618+ (denverton), DSM 7.2, with no-name USB 2.5G adapter.

```
|__usb2          1d6b:0003:0404 09  3.00 5000MBit/s 0mA 1IF  (Linux 4.4.302+ xhci-hcd xHCI Host Controller 0000:00:15.0) hub
  |__2-2         0bda:0411:0104 09  3.00 5000MBit/s 0mA 1IF  (Generic 4-Port USB 3.0 Hub syno.ext.hub) hub
    |__2-2.2     0bda:8156:3100 00  3.20 5000MBit/s 256mA 1IF  (Realtek USB 10/100/1G/2.5G LAN 001000001)

```

**Reply by ektorasdj** [reply]
Re writing my comment to the correct thread: 

"Hello,
I can't begin to thank you for developing these drivers.
DS720+
Club 3D CAC-142
DSM 7.2.4-69057 Update 4
Installed on top of 2.17.1-1 without any errors.
Speeds are full 290mb/s (stable 290mb/s on transfers to NVMe storage volume)
MTU 9000
I have them installed almost 5 days and no issues yet."

**Comment by ParPeter** [comment]
I installed this on a DS923+, a DS920+ and a DS918+.
Three different experiences:
DS923+ - installed and connected straight away
DS920+ - install didn't complete (and connect) until I'd rebooted the DS
DS918+ - install appeared to be following the same pattern as the DS920+, except that it didn't connect, because it had chosen a local IP address, or had one chosen for it, that wasn't on my network. Was using DHCP, should have been on 192.168.1.X but had 192.168.54.X. I had to connect to one of the 1Gb ports to get round it. It's OK now.

**Comment by UNIPART82** [comment]
I'd like to try using 2.5G network for "DS218+" that I'm using because it's not broken yet.


DS218+ = Apollolake


![NAS218-Synology-DiskStation (5)](https://github.com/bb-qq/r8152/assets/162960224/962ba55a-8a84-4441-8f2a-82add88a563e)


r8152-apollolake-2.17.1-1_7.1 : Error


r8152-apollolake-2.17.1-1_6.2 : Unable to install


r8152-apollolake-2.17.1-1_7.2 : Error


![NAS218-Synology-DiskStation (1)](https://github.com/bb-qq/r8152/assets/162960224/f74400db-560e-4f8d-96c9-26c484e06c1c)


![NAS218-Synology-DiskStation (2)](https://github.com/bb-qq/r8152/assets/162960224/c76f0426-97ee-4a67-b4d4-36c6ef0ab3b3)

< This package is provided by third-party developers and is not verified by Synology. If you click I agree, you are solely responsible for any device damage or data loss that may result from using the package. >


![NAS218-Synology-DiskStation (3)](https://github.com/bb-qq/r8152/assets/162960224/7f63e047-cb62-4a67-b13b-2584e06406ca)

< Package installation failed >


![NAS218-Synology-DiskStation (4)](https://github.com/bb-qq/r8152/assets/162960224/1229d717-820f-4a4b-8a87-41b8e16cdf4f)

< Error >


I need your help.

**Reply by MilesTEG1** [reply]
@UNIPART82 
Did you follow those steps:

```
Note:

The installation will fail the first time. After that, run the following command from the SSH terminal:
sudo install -m 4755 -o root -D /var/packages/r8152/target/r8152/spk_su /opt/sbin/spk_su

```

**Reply by UNIPART82** [reply]
Thank you.

I read the article many times, but after proceeding with the part I forgot, it was installed without any problems.


![NAS218-Synology-DiskStation (6)](https://github.com/bb-qq/r8152/assets/162960224/ab7803d4-c383-4626-a82b-66eee8394de1)


![무제-1](https://github.com/bb-qq/r8152/assets/162960224/525406f4-9533-4094-95dd-2fd330f33b5d)

**Reply by MilesTEG1** [reply]
Si, does it works now ?

Maybe you should uninstall the driver , clean everything it installed and reboot. And then installe it again.

**Reply by UNIPART82** [reply]
![NAS218-Synology-DiskStation](https://github.com/bb-qq/r8152/assets/162960224/d1b73e46-5c2a-490c-8dec-6587357df25d)


It's very slow, but it's working.


Now I think I need to buy a . . . 2.5g switching hub. T, . T

**Comment by MilesTEG1** [comment]
Hello,
I installed this version, soon after the pkg for dsm7.2 was available, over the previous version without problem.
But that was just apparent.

Because I experience several problem in DSM with some applications and network transfer.
Some backup failed inexplicably , same for some snapshots. Dsm was very slow to load in navigator... more slow than usual.
At first I didn't suspect those problems to be related to the driver...

But when I realized that  transferring files over smb was very slow, I suspect the driver.
I uninstalled it , cleaned the nas from every file it cames with, reboot and the reinstall the driver from scratch.
And since , no more problems !

That's weird that it caused problem to something unrelated with network transfert...
But it seems that it was my source of problems.

So I suggest you clean everything related to the driver before updating.

@bb-qq Is there a possibility that the driver auto clean previous version before installing itself?

**Reply by bb-qq** [reply]
If the settings are completely deleted when the driver is updated, it would be inconvenient because also be cleared the IP address and other settings.
So I would like to fix the root cause. If the problem occurs during the upgrade, please create a new issue and give me the log information so that I can solve the cause of the problem.

**Comment by ParPeter** [comment]
The prices, of 2.5g switches, are coming down. Slowly, but it is happening.

**Comment by redhook62** [comment]
Works as well !

Tested for backups of several TB of data, 
no problems and no disconnections. 
2.4 Gb constant speed with a TRENDnet TUC-ET2G USB adapter.
DS420+ SSD cache + 6 Go of RAM

**Comment by riiviin** [comment]
DS918+ with a Plugable USBC-E2500
I can get everything to work with MTU at 9000 except my NFS shares (32KB read/write packet size). The shares will immediately start working without any other changes or restarting the NAS or my server if I drop MTU back down to 1500. Speeds with MTU set to 1500 are hitting 2.5Gbps, so this is not a big deal.

**Comment by cosminmocan** [comment]
This is my first time trying this driver so I cannot say if it's better or worse, while the nfs server works wonders ( i use with with proxmox ), with iscsi i have errors, as seen below:

My nas is a synology ds923+ , dsm 7.2.1
<details>
<summary>iSCSi dmesg error</summary>
<br>
iSCSI:iscsi_target.c:4612:stop_syno_worker_for_cmds kref_get_unless_zero(ffff880033e1fe10) falied!
iSCSI_F:iscsi_target.c:4780:iscsit_close_connection Close - I[iqn.1993-08.org.debian:01:e85accf3c75][10.10.10.99:58254], T[(null)]
iSCSI:iscsi_target_parameters.c:39:iscsi_login_rx_data rx_data returned 0, expecting 48.
iSCSI_F:iscsi_target_login.c:1303:iscsi_target_login_sess_out iSCSI Login negotiation failed - I[][10.10.10.99:58266], T[][10.10.10.95:3260]
iSCSI:iscsi_target_parameters.c:39:iscsi_login_rx_data rx_data returned 0, expecting 48.
iSCSI_F:iscsi_target_login.c:800:iscsi_post_login_handler Discovery session from I[iqn.1993-08.org.debian:01:e85accf3c75][10.10.10.99:58278]
iSCSI_F:iscsi_target_login.c:1303:iscsi_target_login_sess_out iSCSI Login negotiation failed - I[][10.10.10.99:58270], T[][10.10.10.95:3260]
iSCSI:iscsi_target.c:4612:stop_syno_worker_for_cmds kref_get_unless_zero(ffff880104bd7e10) falied!
iSCSI_F:iscsi_target.c:4780:iscsit_close_connection Close - I[iqn.1993-08.org.debian:01:e85accf3c75][10.10.10.99:58278], T[(null)]
</details>

There is a direct connection between the nas and the proxmox device, both use a usb 2.5 nic from aliexpress
`0bda:8156:3104 00  3.20 5000MBit/s 256mA 1IF  (Realtek USB 10/100/1G/2.5G LAN 4013000001)`

Could the older driver be better for my use case (iscsi), or is this even suppose to work with iscsi in the first place ?
Regards and thank you for this awesome project 💯

**Comment by dreimer1986** [comment]
Confirmed working on:
- DS420+ (Geminilake) with DSM7.2 
- 2x DS414 (ArmadaXP) with DSM7.1

Made a BIG Backup of the DS420+ over a script running on my 2.5G connected Windows PC to split the data almost in half to fit on the two old NAS systems. The old ones are too slow to really reach a limit on the PC connection, but still it's faster than a 1G connection. (~80 MB/s vs about 65 MB/s)
Never tried the Jumboframe idea for more speed tbh. Every device seems to interpret the frame size differently and whatever I tried I failed badly to get a reliable connection between the NAS and my Windows devices...

**Comment by ziv01** [comment]
Not working on my DS220J will complete istallation but will not start. 2.17.1.-1 is OK but keep disconnectiong

UPDATE: managed to install it , the NIC shows up but in a minute it disconnect.

**Reply by bb-qq** [reply]
Please refer this issue: https://github.com/bb-qq/r8152/issues/275

**Comment by NeoXTof** [comment]
Tested on my DS1520+ (dsm 7.2.1 with 20GB ram) with a noname from aliexpress. Can not get MTU 9000 stable for more than 1 hour, so I moved back to 1500.

**Comment by PatriceFo** [comment]
Hello
first of all, nice work, it works wonderfully on a recent NAS DS224+ with a 7.2.1.

I was wondering if there was any chance that there was an asset compatible with the DS212+ on 6.2.4 , I can't really find what type of mic is present in this sort of old thing.
Thx

**Reply by NeoXTof** [reply]
![image](https://github.com/bb-qq/r8152/assets/163831075/269f52fc-627c-4240-89da-b62ddfa885b3)

**Reply by PatriceFo** [reply]
thx, does that mean that i should search for a :
r8152-8876281XXX-2_6.2.spk ?

If so, cant find that kind of stuff in the asset list :-(

**Reply by arigon** [reply]
> Hello first of all, nice work, it works wonderfully on a recent NAS DS224+ with a 7.2.1.
> 
> I was wondering if there was any chance that there was an asset compatible with the DS212+ on 6.2.4 , I can't really find what type of mic is present in this sort of old thing. Thx

Which USB dongle are you using?

**Reply by PatriceFo** [reply]
> > Hello first of all, nice work, it works wonderfully on a recent NAS DS224+ with a 7.2.1.
> > I was wondering if there was any chance that there was an asset compatible with the DS212+ on 6.2.4 , I can't really find what type of mic is present in this sort of old thing. Thx
> 
> Which USB dongle are you using?
A Realtek RTL8152/RTL8153/RTL8156 chip or so on a Trendnet TUC-ET2G usb dongle for the Ds224+. For the DS212, I still can't find a library compatible with the old processor and its old version of the Nas operating system.

**Reply by arigon** [reply]
Thank you! I'm replacing the old DS216+II now with the DS224+ because I couldn't get the UGREEN working with the DS216+II without killing the UPS USB connection. I tried the last few r8152 versions and no luck so far. So I hope the DS224+ will work fine. :)

**Reply by Sjekke** [reply]
I have the same problem with a 916+ and Ugreen adapter Realtek RTL8156BG It works but when transferring a big file the adapter suddenly stops transmitting. With a Pluggable adapter I don't have this problem. I hope bb-qq can solve the problem...

**Reply by arigon** [reply]
> I have the same problem with a 916+ and Ugreen adapter Realtek RTL8156BG It works but when transferring a big file the adapter suddenly stops transmitting. With a Pluggable adapter I don't have this problem. I hope bb-qq can solve the problem...

This might be the same as #415?

**Comment by mrjohnpoz** [comment]
Great release.. Confirmed working with ds918+ on dsm 7.2.1u5 using CLUB 3D CAC-1420, and same CAC-1420 on windows 10 using realtek driver 10.61.20.327

Also able to use jumbo 9000 vs 1500 now.. Seems stable with testing with iperf3 and file copies.. Little bit of a boost to be honest.. Nice!!

Before change to 9000 mtu

[  5]   0.00-10.02  sec  2.76 GBytes  2.37 Gbits/sec

After

[  5]   0.00-10.02  sec  2.89 GBytes  2.48 Gbits/sec

**Comment by WolfAGR** [comment]
Confirmed working on DS414 DSM 7.1.1-42962 Update 8 with WavelinkWL-NWU340G @5Gbe on 10Gbe switch.

**Reply by WolfAGR** [reply]
After reboot I lost the network.
